File: 프로그래머스/2/42842. 카펫/카펫.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(int brown, int yellow) {
    vector<int> answer(2);
    int area = brown + yellow;
    int height, width;
    
    for (height = 3; height < area / 2; height++)
    {
        if(area % height == 0)
        {
            width = area / height;
            if(2 * (width+height) - 4 == brown)
            {
                answer.front() = width;
                answer.back() = height;
                break;
            }
        }
    }
    
    return answer;
}

File: 프로그래머스/2/42842. 카펫/카펫.py
import math
def solution(brown, yellow):
    answer = []
    hap = brown + yellow
    
    for i in range(2, hap//2):
        if (i-2) * ((hap//i) - 2) == yellow:
            answer.append(hap//i)
            answer.append(i)
            break
    
    return answer